Proposals/0007-swift-subprocess.md

```diff
@@ -286,92 +459,305 @@ Both styles of the `run` methods provide convenient overloads that allow develop
+    @available(FoundationPreview 0.4, *)
+    @available(iOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    public struct ProcessIdentifier: Sendable, Hashable, Codable {
+        /// Windows specifc process identifier value
+        public let processID: DWORD
+        /// Windows specific thread identifier associated with process
+        public let threadID: DWORD
+
+        public init(processID: DWORD, threadID: DWORD)
+    }
+}
+#endif // canImport(WinSDK)
+
+extension Subprocess.ProcessIdentifier : CustomStringConvertible, CustomDebugStringConvertible {}
 ```
 
+#### Unmanaged Subprocess
 
-### Signals
+In addition to the managed `run` family of methods, `Subprocess` also supports an unmanaged `runDetached` method that simply spawns the executable and returns its process identifier without awaiting for it to complete. This mode is particularly useful in scripting scenarios where the subprocess being launched requires outlasting the parent process. This setup is essential for programs that function as “trampolines” (e.g., JVM Launcher) to spawn other processes.
 
-`Subprocess` uses `struct Subprocess.Signal` to represent the signal that could be sent via `sendSignal()`. Developers could either initialize `Signal` directly using the raw signal value or use one of the common values defined as static property.
+Since `Subprocess` is unable to monitor the state of the subprocess or capture and clean up input/output, it requires explicit `FileDescriptor` to bind to the subprocess’ IOs. Developers are responsible for managing the creation and lifetime of the provided file descriptor; if no file descriptor is specified, `Subprocess` binds its standard IOs to `/dev/null`.
 
 ```swift
+@available(FoundationPreview 0.4, *)
+@available(iOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
 extension Subprocess {
+    /// Run a executable with given parameters and return its process
+    /// identifier immediately without monitoring the state of the
+    /// subprocess nor waiting until it exits.
+    ///
+    /// This method is useful for launching subprocesses that outlive their
+    /// parents (for example, daemons and trampolines).
+    ///
+    /// - Parameters:
+    ///   - executable: The executable to run.
+    ///   - arguments: The arguments to pass to the executable.
+    ///   - environment: The environment to use for the process.
+    ///   - workingDirectory: The working directory for the process.
+    ///   - platformOptions: The platform specific options to use for the process.
+    ///   - input: A file descriptor to bind to the subprocess' standard input.
+    ///   - output: A file descriptor to bind to the subprocess' standard output.
+    ///   - error: A file descriptor to bind to the subprocess' standard error.
+    /// - Returns: the process identifier for the subprocess.
+    public static func runDetached(
```

Comment:
Is this in the end just spawning an unstructured Task and calling `run` inside that task? If that's the case I would really prefer we don't offer that API and just expect users to construct that unstructured task themselves. This will allow them to still have some ownership of the task and cancel it if needed.

Reply:
No there is no unstructured task. This API is not `async` at all. It simply spawns the process and returns the PID to you as if you were calling `posix_spawn` yourself.